Doc: Add description of morphometrics columns headings

mariehbourget opened the pull request:
## Checklist

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [ ] I've linked relevant issues in the PR body
- [x] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [ ] I've added relevant tests for my contribution
- [x] I've updated the documentation and/or added correct docstrings
- [x] I've assigned a reviewer
- [x] I've consulted [ADS's internal developer documentation](https://github.com/neuropoly/axondeepseg/wiki/Guidelines-for-contribution) to ensure my contribution is in line with any relevant design decisions

## Description
This PR adds a description of the columns headings of the `axon_morphometrics` file produces by ADS.
`solidity`, `eccentricity` and `orientation` descriptions are taken directly from the descriptions of `skimage measure regionprops`.

jcohenadad commented on docs/source/documentation.rst:
```diff
@@ -365,6 +365,49 @@ This will generate **'axon_morphometrics.xlsx'** file in each of folders::
+The resulting **'axon_morphometrics'** file will contain the following columns headings:
+
+x0
+                    Axon X centroid position in pixels. Extracted with `skimage.measure.regionprops <scikit-image https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops>`_.
+
+y0
+                    Axon Y centroid position in pixels. Extracted with `skimage.measure.regionprops <scikit-image https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops>`_.
+
+gratio
+                    Ratio between the axon equivalent diameter and the axon+myelin (fiber) equivalent diameter.
+
+axon_area
+                    Area of the axon region in :math:`{\mu}`\ m\ :sup:`2`\ . Extracted with `skimage.measure.regionprops <scikit-image https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops>`_.
+
+axon_perimeter
+                    Perimeter of the axon object in :math:`{\mu}`\ m. Extracted with `skimage.measure.regionprops <scikit-image https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops>`_.
+
+myelin_area
+                    Difference between axon+myelin (fiber) area and axon area in :math:`{\mu}`\ m\ :sup:`2`\ .
+
+axon_diameter
+                    Equivalent diameter of the axon in :math:`{\mu}`\ m (diameter of a circle with the same area as the region). Extracted with `skimage.measure.regionprops <scikit-image https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops>`_.
+
+myelin_thickness
+                    Half of the difference between the axon+myelin (fiber) diameter and the axon diameter in :math:`{\mu}`\ m.
+
+axonmyelin_area
+                    Area of the axon+myelin (fiber) region in :math:`{\mu}`\ m\ :sup:`2`\ . Extracted with `skimage.measure.regionprops <scikit-image https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops>`_.
+
+axonmyelin_perimeter
+                    Perimeter of the axon+myelin (fiber) object in :math:`{\mu}`\ m. Extracted with `skimage.measure.regionprops <scikit-image https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops>`_.
+
+solidity
+                    Ratio of pixels in the axon region to pixels of the convex hull image. Extracted with `skimage.measure.regionprops <scikit-image https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops>`_.
+
+eccentricity
+                    Eccentricity of the ellipse that has the same second-moments as the axon region. Extracted with `skimage.measure.regionprops <scikit-image https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops>`_.
+
+orientation
+                    Angle between the 0th axis (rows) and the major axis of the ellipse that has the same second moments as the axon region. Extracted with `skimage.measure.regionprops <scikit-image https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops>`_.
```
```suggestion
.. list-table:: axon_morphometrics.csv/xls
   :widths: 20 80
   :header-rows: 1

   * - Field
     - Description
     - Heading row 1, column 3
   * - x0
     - Axon X centroid position in pixels. Extracted with `skimage.measure.regionprops <scikit-image https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops>`_.
   * - y0
     - Axon Y centroid position in pixels. Extracted with `skimage.measure.regionprops <scikit-image https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops>`_.
   * - gratio
     - Ratio between the axon equivalent diameter and the axon+myelin (fiber) equivalent diameter.
   * - axon_area
     - Area of the axon region in :math:`{\mu}`\ m\ :sup:`2`\ . Extracted with `skimage.measure.regionprops <scikit-image https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops>`_.
   * - axon_perimeter
     - Perimeter of the axon object in :math:`{\mu}`\ m. Extracted with `skimage.measure.regionprops <scikit-image https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops>`_.
   * - myelin_area
     - Difference between axon+myelin (fiber) area and axon area in :math:`{\mu}`\ m\ :sup:`2`\ .
   * - axon_diameter
     - Equivalent diameter of the axon in :math:`{\mu}`\ m (diameter of a circle with the same area as the region). Extracted with `skimage.measure.regionprops <scikit-image https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops>`_.
   * - myelin_thickness
     - Half of the difference between the axon+myelin (fiber) diameter and the axon diameter in :math:`{\mu}`\ m.
   * - axonmyelin_area
     - Area of the axon+myelin (fiber) region in :math:`{\mu}`\ m\ :sup:`2`\ . Extracted with `skimage.measure.regionprops <scikit-image https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops>`_.
   * - axonmyelin_perimeter
     - Perimeter of the axon+myelin (fiber) object in :math:`{\mu}`\ m. Extracted with `skimage.measure.regionprops <scikit-image https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops>`_.
   * - solidity
     - Ratio of pixels in the axon region to pixels of the convex hull image. Extracted with `skimage.measure.regionprops <scikit-image https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops>`_.
   * - eccentricity
     - Eccentricity of the ellipse that has the same second-moments as the axon region. Extracted with `skimage.measure.regionprops <scikit-image https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops>`_.
   * - orientation
     - Angle between the 0th axis (rows) and the major axis of the ellipse that has the same second moments as the axon region. Extracted with `skimage.measure.regionprops <scikit-image https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops>`_.
```
Maybe it would look nicer as a table

vs74 replied:
I agree with @jcohenadad  that a table would be a much better option.

vs74 commented on docs/source/documentation.rst:
```diff
@@ -365,6 +365,49 @@ This will generate **'axon_morphometrics.xlsx'** file in each of folders::
+                    Axon Y centroid position in pixels. Extracted with `skimage.measure.regionprops <scikit-image https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops>`_.
+
+gratio
+                    Ratio between the axon equivalent diameter and the axon+myelin (fiber) equivalent diameter.
```
Perhaps we can skip adding the formula, but it would be very important to mention that gratio is computed using the area of the axon and myelin objects. `gratio = axon area / (axon area + myelin area)`

mariehbourget replied:
I agree, I think the formula is too much but I will add a mention that the "equivalent diameter" is based on the area.

Also, the g-ratio is the inner/outer diameter (`gratio = axon_diameter/aonmyelin_diameter`), so the formula with the area would have a sqrt, no? `gratio = sqrt(axon area / (axon area + myelin area))`.

jcohenadad replied:
yes it would be great to add a formula! @mathieuboudreau can confirm what is the correct equation

vs74 replied:
> Also, the g-ratio is the inner/outer diameter (`gratio = axon_diameter/aonmyelin_diameter`), so the formula with the area would have a sqrt, no? `gratio = sqrt(axon area / (axon area + myelin area))`.

Yes, you are correct, it is `gratio = sqrt(axon area / (axon area + myelin area))`

mariehbourget replied:
Should I go with the general definition `gratio = axon_diameter/axonmyelin_diameter`, or with what ADS actually do `np.sqrt(axon_area / axonmyelin_area)` ?

vs74 replied:
Although it doesn't matter whichever formula we use as the result is the same, I'm favor of the actual formula we do in ADS, that is, `np.sqrt(axon_area / axonmyelin_area)`. @mathieuboudreau and @jcohenadad can chime in as well.

mariehbourget replied:
@jcohenadad, @vs74
We need an agreement on the g-ratio definition. I wrote "in words" the general definition based on the ratio of **diameters**. But we want to add a formula, and the ADS formula is the square root of the **areas** ratio.  I think both the definition and formula should refer to the same metric for clarity, either diameter or area.

Currently in ADS, both definitions gives the same results, because the diameter is computed with the area of a circle.
However, at some point, we will add the option to compute the diameter as the minor axis of an ellipse as well. In that case, the current g-ratio with areas in ADS will have to change for the definition based on diameters. This is why I prefer the diameter definition.

For this PR, I see 2 options, which one should we use?

**Option 1, diameter:**
Ratio between the axon equivalent diameter and the axon+myelin (fiber) equivalent diameter.
`gratio = axon_diameter/axonmyelin_diameter`

**Option 2, area:**
Square root of the ratio between the axon area and the axon+myelin (fiber) area.
`gratio = sqrt(axon_area / axonmyelin_area)`

mathieuboudreau replied:
**I am firmly voting for Option 1.** It's important that we use the same definition (equation) as the literature (which is that gratio = axon_diameter/axonmyelin_diameter), but then as you've done we should specify that because this is only valid for circles (ideal case), we're calculating the diameters from equivalent circles of the same are.

vs74 replied:
The way how current g-ratio would definitely change with the inclusion of minor axis of an ellipse to compute diameter and we would change the g-ratio to generalized definition(**option 1**) rather than what is implemented in ADS now ( using **option 2**)

Hence, I am also in favor of **option 1**, a generalized definition.

jcohenadad replied:
+1 for option 1

mariehbourget replied:
Thank you for your feedback!
Fixed in 055c9d6.

mariehbourget commented on docs/source/documentation.rst:
```diff
@@ -365,6 +365,49 @@ This will generate **'axon_morphometrics.xlsx'** file in each of folders::
+                    Ratio between the axon equivalent diameter and the axon+myelin (fiber) equivalent diameter.
+
+axon_area
+                    Area of the axon region in :math:`{\mu}`\ m\ :sup:`2`\ . Extracted with `skimage.measure.regionprops <scikit-image https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops>`_.
```
According to `regionprops`, the area is directly the number of pixels of the region, which we convert to um with the pixelsize. It's the diameter (equivalent diameter) that is computed as an "diameter of a circle with the same area as the region".

mariehbourget commented on docs/source/documentation.rst:
```diff
@@ -365,6 +365,49 @@ This will generate **'axon_morphometrics.xlsx'** file in each of folders::
+myelin_thickness
+                    Half of the difference between the axon+myelin (fiber) diameter and the axon diameter in :math:`{\mu}`\ m.
+
+axonmyelin_area
```
Yes I agree, like the gratio, I'll make sure that we mentioned the diameter is based on the area of a circle.

jcohenadad reviewed (commented):
for some reasons the table i added does not show up:
![image](https://user-images.githubusercontent.com/2482071/119150141-29ea3e80-ba1c-11eb-8b52-71b7ead678a7.png)

mariehbourget commented:
> for some reasons the table I added does not show up:

I'm fixing it now, along with the others changes, no problem.
Thank you for the table, it's so much better!

mariehbourget commented:
The table is updated if you want to double-check.
I just need feedback on the g-ratio description.

Question: regionprops extracts the values in pixels, should I mentioned that the convert them um and um2?
example for axon area:
`Area of the axon region in um2 . Extracted with skimage.measure.regionprops in pixels and converted to um2 according to image pixel size.`

jcohenadad commented on docs/source/documentation.rst:
```diff
@@ -365,6 +365,43 @@ This will generate **'axon_morphometrics.xlsx'** file in each of folders::
+Morphometrics file
+~~~~~~~~~~~~~~~~~~
+
+The resulting **'axon_morphometrics'** file will contain the following columns headings:
```
```suggestion
The resulting **'axon_morphometrics'** file will contain the following columns headings. Most of the metrics are computed using `skimage.measure.regionprops <https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops>`_.
```
I would remove all the "Extracted with `skimage.measure.regionprops <https://scikit-image.org/docs/stable/api/skimage.measure.html#regionprops>`_." in the table (it appears almost at every row-- this duplication of information is unnecessary and distracting)-- and just put it there once.

vs74 commented:
> Question: regionprops extracts the values in pixels, should I mentioned that we convert them um and um2?
> example for axon area:
> `Area of the axon region in um2 . Extracted with skimage.measure.regionprops in pixels and converted to um2 according to image pixel size.`

I guess it not needed to go into such details.

mariehbourget commented:
Hi everyone! The requested changes are made.
The only thing missing is an agreement on the gratio definition [here](https://github.com/neuropoly/axondeepseg/pull/516#discussion_r636702174).
Thanks!

mariehbourget commented:
The PR is ready for re-review, thanks!

vs74 reviewed (approved):
LGTM. Thanks, @mariehbourget for this important contribution.
